db/background_mgr_resync_dcp_test.go

```diff
@@ -444,6 +445,93 @@ function sync(doc, oldDoc){
 	return db, ctx
 }
 
+// TestResyncMou ensures that resync updates create mou, and preserve pcas in mou in the case where resync is reprocessing an import
+func TestResyncMou(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Test requires Couchbase Server")
+	}
+
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyMigrate, base.KeyImport)
+	db, ctx := setupTestDBWithOptionsAndImport(t, nil, DatabaseContextOptions{})
+	defer db.Close(ctx)
+
+	if !db.Bucket.IsSupported(sgbucket.BucketStoreFeatureMultiXattrSubdocOperations) {
+		t.Skip("Test requires multi-xattr subdoc operations, CBS 7.6 or higher")
+	}
+
+	db.Options.QueryPaginationLimit = 100 // Required for principal ID query to not deadlock
+
+	initialImportCount := db.DbStats.SharedBucketImport().ImportCount.Value()
+	collection := GetSingleDatabaseCollectionWithUser(t, db)
+	docBody := Body{"foo": "bar"}
+
+	// Create a document via SGW.  mou should not be updated
+	_, doc, err := collection.Put(ctx, "sgWrite", docBody)
+	require.NoError(t, err)
+	sgWriteCas := doc.Cas
+
+	syncData, mou, _ := getSyncAndMou(t, collection, "sgWrite")
+	require.NotNil(t, syncData)
+	require.Nil(t, mou)
+
+	// 2. Create via the SDK
+	_, err = collection.dataStore.WriteCas("sdkWrite", 0, 0, docBody, 0)
+	require.NoError(t, err)
+
+	base.RequireWaitForStat(t, func() int64 {
+		return db.DbStats.SharedBucketImport().ImportCount.Value()
+	}, initialImportCount+1)
+
+	syncData, initialSDKMou, _ := getSyncAndMou(t, collection, "sdkWrite")
+	require.NotNil(t, syncData)
+	require.NotNil(t, initialSDKMou)
+
+	// Update sync function
+	syncFn := `
+function sync(doc, oldDoc){
+	channel("resync_channel");
+}`
+	_, err = collection.UpdateSyncFun(ctx, syncFn)
+	require.NoError(t, err)
+
+	resyncMgr := NewResyncManagerDCP(db.MetadataStore, base.TestUseXattrs(), db.MetadataKeys)
+	require.NotNil(t, resyncMgr)
+
+	initialStats := getResyncStats(resyncMgr.Process)
+	log.Printf("initialStats: processed[%v] changed[%v]", initialStats.DocsProcessed, initialStats.DocsChanged)
+
+	options := map[string]interface{}{
+		"database":            db,
+		"regenerateSequences": false,
+		"collections":         ResyncCollections{},
+	}
+
+	err = resyncMgr.Start(ctx, options)
+	require.NoError(t, err)
+
+	err = WaitForConditionWithOptions(t, func() bool {
+		var status BackgroundManagerStatus
+		rawStatus, _ := resyncMgr.GetStatus(ctx)
+		_ = json.Unmarshal(rawStatus, &status)
+		return status.State == BackgroundProcessStateCompleted
+	}, 200, 200)
```

Comment:
just to make this easier to debug for the future, so it will print the status if it does fail:

```suggestion
	require.EventuallyWithT(t, func(c *assert.CollectT) {
		var status BackgroundManagerStatus
		rawStatus, err := resyncMgr.GetStatus(ctx)
		assert.NoError(t, err)
		assert.Equal(t, json.Unmarshal(rawStatus, &status)
		assert.Equal(t, BackgroundProcessStateComplete,  status.State)
	}, 40 * time.Second, 200 * time.Millisecond)
```